packages/dashboard-core-plugins/src/panels/ColumnSelectionValidator.ts

```diff
@@ -0,0 +1,7 @@
+import { PanelComponent } from '@deephaven/dashboard';
+import { LinkColumn } from '../linker/LinkerUtils';
+
+export type ColumnSelectionValidator = (
+  panel: PanelComponent,
+  tableColumn?: Partial<LinkColumn>
```

Comment:
I think instead of using `Partial<LinkColumn>`, you want to define a new type, `LinkColumnSelection = { name: string, type?: string }` and use that.

With `Partial`, you then also have to check that name is defined, which it always will be.